01/main.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+)
+
+func main() {
+	// open the file
+	file, err := os.Open("input")
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	// read line by line
+	scan := bufio.NewScanner(file)
+	depth := 0
+	counter := 0
+	for scan.Scan() {
+		newDepth, err := strconv.Atoi(scan.Text())
+		if err != nil {
+			fmt.Printf("Error converting text to int. %s\n", err)
+		}
+
+		if depth == 0 {
+			fmt.Printf("%d (N/A - no previous measurement)\n", newDepth)
+		} else {
+			op := "(decreased)"
+			if newDepth > depth {
+				op = "(increased)"
+				counter++
+			}
+			fmt.Printf("%d %s\n", newDepth, op)
+		}
+
+		depth = newDepth
+	}
+
+	// check for errors
+	if err := scan.Err(); err != nil {
+		fmt.Printf("Scanner contained an error: %s\n", err)
```

Comment:
shouldn't the program stop if we could not read everything?

Reply:
yep, that was the first exercise so I had too much debug added. I'll remove it and make it simpler